### DISCUSSION TITLE
Is an implementation of "Encrypted Client Hello" possible in the future?

### DISCUSSION THREAD

**Discussion by tigernero79** [DNSCrypt/dnscrypt-proxy#1831] [Q&A]
At this link I read that cloudflare and firefox are abandoning esni in favor of ECH "Encrypted Client Hello" 
for reasons of better privacy, would this be possible to implement it locally as esni?

referral link:
https://blog.mozilla.org/security/2021/01/07/encrypted-client-hello-the-future-of-esni-in-firefox/

**Comment by jedisct1** [comment]
https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Local-DoH
